### ISSUE TITLE
 Unable to connect to Dataverse: Unable to cast object of type 'generatedProxy_3' to type 'Microsoft.PowerPlatform.Dataverse.Client.IOrganizationServiceAsync'.

### ISSUE THREAD

**Issue by twiga2013** [microsoft/PowerPlatform-DataverseServiceClient#295]
Hi
On version 1.0.1 of Microsoft.PowerPlatform.Dataverse.Client
When creating records in parallel we are getting the following exception.
Unable to connect to Dataverse: Unable to cast object of type 'generatedProxy_3' to type 'Microsoft.PowerPlatform.Dataverse.Client.IOrganizationServiceAsync'.
Source: System.Private.CoreLib
Method: ChkCastAny
DateUTC: 7/15/2022
TimeUTC: 3:25:35 PM
Error: Unable to cast object of type 'generatedProxy_3' to type 'Microsoft.PowerPlatform.Dataverse.Client.IOrganizationServiceAsync'.
HelpLink Url: Not Provided
Stack Trace: at System.Reflection.DispatchProxy.Create[T,TProxy]()
   at System.ServiceModel.Channels.ServiceChannelProxy.CreateProxy[TChannel](MessageDirection direction, ServiceChannel serviceChannel)
   at System.ServiceModel.Channels.ServiceChannelFactory.CreateProxy[TChannel](MessageDirection direction, ServiceChannel serviceChannel)
   at System.ServiceModel.Channels.ServiceChannelFactory.CreateChannel[TChannel](EndpointAddress address, Uri via)
   at System.ServiceModel.ChannelFactory`1.CreateChannel(EndpointAddress address, Uri via)
   at System.ServiceModel.ChannelFactory`1.CreateChannel()
   at System.ServiceModel.ClientBase`1.CreateChannel()
   at System.ServiceModel.ClientBase`1.CreateChannelInternal()
   at System.ServiceModel.ClientBase`1.get_Channel()
   at System.ServiceModel.ClientBase`1.get_InnerChannel()
   at Microsoft.PowerPlatform.Dataverse.Client.ConnectionService.ConnectAndInitServiceAsync(OrganizationDetail orgdata, Boolean IsOnPrem, Uri homeRealmUri)
   at Microsoft.PowerPlatform.Dataverse.Client.ConnectionService.DoDirectLoginAsync(Boolean IsOnPrem)
   at Microsoft.PowerPlatform.Dataverse.Client.ConnectionService.InitServiceAsync()
======================================================================================================================

**Comment by MattB-msft** [comment]
Hi @twiga2013 ,  Are you running this on an Azure function? and if so is this a constant repro? or is it happening randomly? 

thanks

**Comment by MattB-msft** [comment]
Ok,  this is a duplicate of #259  Take a look there..  We are fairly sure this has to do with scale out on an Azure Function.
The team is still digging though it.

**Comment by twiga2013** [comment]
Hi @MattB-msft is there any progress on the issue.

**Comment by MattB-msft** [comment]
We are discussing the behavior with the Azure Functions team and the .net core team and are working though various possible solutions, including some targeted fixes in our client. 

We are not the only team experiencing this issue.

Sorry about the time its taking to work this though.

**Comment by twiga2013** [comment]
Hi @MattB-msft thanks for responding. As the load increases and Azure Function Scales out we are seeing different kinds of errors been thrown from
1. System.Net.Sockets.SocketExceptionSystem
2. WCF errors
3. System.InvalidCastException

What is the recommendation on UseUniqueInstance parameter on the SerivceClient constructor?

**Comment by MattB-msft** [comment]
UseUniqueInstance = true will cause the client to create more instances of itself, which is more memory intensive and less performant.

**Comment by MattB-msft** [comment]
@twiga2013  can you please add your comments to the linked issue,, trying to keep it all in one place.

Going to close this issue here, Please add comments to #259
